app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php

```diff
@@ -636,6 +640,36 @@ private function cleanupDisabledJobs($groupId)
         }
     }
 
+    /**
+     * Cleanup jobs that were left in a running state due to an unexpected stop
+     *
+     * @param string $groupId
+     * @return void
+     */
+    private function cleanupRunningJobs($groupId)
+    {
+        $scheduleResource = $this->_scheduleFactory->create()->getResource();
+        $connection = $scheduleResource->getConnection();
+
+        $jobs = $this->_config->getJobs();
+        foreach (array_keys($jobs[$groupId]) as $jobCode) {
+            $connection->update(
+                $scheduleResource->getMainTable(),
+                [
+                    'status' => \Magento\Cron\Model\Schedule::STATUS_ERROR,
+                    'messages' => 'Time out'
+                ],
+                $connection->quoteInto(
+                    'job_code = ? ' .
+                    'AND status = ? ' .
+                    'AND (executed_at < UTC_TIMESTAMP() - INTERVAL 1 DAY)',
```

Comment:
We should probably make this "1 day" interval configurable in a new field in the backend of Magento per cron group?
Maybe there are cronjob processes out there which take more then 1 day to complete, you never know...